plotly_resampler/figure_resampler/figure_resampler_interface.py

```diff
@@ -705,20 +705,20 @@ def _parse_get_trace_props(
             # timezones in the data
             if len(hf_x) and hf_x.dtype == "object" and isinstance(hf_x[0], (pd.Timestamp, datetime.datetime)):
                 # Assumes that all values in hf_x are either pd.Timestamp or datetime.datetime
-                if any(
-                    x.tzinfo is not None and x.tzinfo != hf_x[0].tzinfo
-                    for x in hf_x[1:]
-                ):
-                    # Remove the timezone data for plotting when multiple timezones
+                try:
+                    hf_x = pd.to_datetime(hf_x, utc=False)
+                except ValueError:
+                    # ValueError will be thrown when there are multiple timezones in the data
+                    # => remove the timezone data for plotting when multiple timezones
                     warnings.warn(
                         "x-data of multiple timezones / fixedoffsets is passed, "
                         + "omitting the timezone data for plotting\n"
                         + "If you are dealing with daylight savings time we suggest converting to one and the same timezone.",
                         UserWarning,
                     )
-                    hf_x = np.asarray(
-                        list(map(lambda x: x.replace(tzinfo=None), hf_x))
-                    )
+
+                    hf_x = [x.replace(tzinfo=None) for x in hf_x]
```

Comment:
I was wondering if this cannot be performed with a pandas (optimized) function. Something like `x.tz_replace(None)`, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.tz_localize.html